src/test_l1_dsVideoDevice.c

```diff
@@ -1126,26 +1141,31 @@ void test_l1_dsVideoDevice_positive_dsSetFRFMode(void)
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     // Define variables
     int result;
-    int index = 0;
     intptr_t handle = -1;
     int validFramerate = 60; // This is just a placeholder value; replace with an appropriate value as per your environment.
 
     // Step 01: Initialize video devices
     result = dsVideoDeviceInit();
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Step 02: Obtain video device handle
-    result = dsGetVideoDevice(index, &handle);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-    UT_ASSERT_EQUAL((handle >= 0), true);
-
-    // Step 03: Set the FRF mode using a valid framerate
-    result = dsSetFRFMode(handle, validFramerate);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    for(int i = 0; i < gDSvideoDevice_NumVideoDevices; i++){
+        result = dsGetVideoDevice(i, &handle);
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
+        UT_ASSERT_EQUAL((handle >= 0), true);
+
+       // Step 03: Set the FRF mode using a valid framerate
+       result = dsSetFRFMode(handle, validFramerate);
+       if(gSourceType == 1) {
+           UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
+       } else if(gSourceType == 0) {
+           UT_ASSERT_EQUAL(result, dsERR_NONE);
```

Comment:
try to set min  to max values 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAVDTypes.h#L497

Reply:
done